plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -724,6 +730,27 @@ public static OptionalInt parseVersion(String metadataLocation)
         }
     }
 
+    /**
+     * This method parses the version from metadata file which is generated while creating
+     * Hadoop path-based iceberg tables (eg: /user/hive/warehouse/db/test_table/metadata/v1.metadata.json).
+     */
+    private static OptionalInt parseHadoopTableMetadataFileVersion(String metadataFileName)
```

Comment:
Consider using a slightly adapted version of what Iceberg does for parsing the version number:

https://github.com/apache/iceberg/blob/1d88e80bacbcb689e7f97932b25bf11b3633b190/core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java#L274C1-L285